source/agora/common/crypto/Schnorr.d

```diff
@@ -64,6 +64,481 @@ nothrow @nogc unittest
     assert(verify(kp.V, signature, "Hello world"));
 }
 
+/// Returns: The accumulated values
+/// Note: be aware of accumulators, when we add to a value we must
+/// ensure the value was initialized with an accumulator (not T.init!)
+T combine (T)(T[] elems)
+    if (is(T == Scalar) || is(T == Point))
+{
+    assert(elems.length > 0);
+
+    if (elems.length == 1)
+        return elems[0];
+    else
+        return sum(elems[1 .. $], elems[0]);  // proper accumulator, not T.init!
+}
+
+/// Returns: The accumulated values
+/// Note: be aware of accumulators, when we add to a value we must
+/// ensure the value was initialized with an accumulator (not T.init!)
+T combine (T)(T lhs, T rhs)
```

Comment:
I don't quite understand the need for `combine` ? We have operator overload and the array version doesn't seem to be used.

Reply:
> the array version doesn't seem to be used.

Indeed, it's dead code now. I'll remove it.

Reply:
Ping

Reply:
The issue is specifically for `Point`, I'll remove the calls to `combine` for `Scalar` as they work ok. The issue is:

```D
unittest
{
    Point p1 = Scalar.random().toPoint();
    Point p2 = Scalar.random().toPoint();

    Point sum1 = p1 + p2;

    // the correct way is to initialize it to p1, and then add the other points
    Point sum2;
    foreach (point; [p1, p2])
        sum2 = sum2 + point;

    assert(sum2 == sum1);  // fails
}
```

It doesn't make sense to add to a point that has been default-initialized. So I wrote a helper function. I could just write this code inline without the function I guess, but I wanted to make this edge-case visible, because it seems to me like we could introduce bugs if we're not careful with using `Point.init`.